### PR TITLE
Ft2023 340: Adding code for custom hooks and hooks implementation.

### PR DESCRIPTION
**Pull request by Souvik-Inno**
Added Custom Hook to count number of times user visited a node.

### CODE REVIEW

File: web/modules/custom/hello_user/src/CurrentUser.php [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  /**
+   * Gets the current user's display name.
+   * 
+   * @return string
+   *   The display name.
+   */
+  public function getUserDisplayName() {
+    return $this->currentUser->getDisplayName();
```
Comment (vivekpan27) [inline web/modules/custom/hello_user/src/CurrentUser.php:36]: Why create a new function for funtionality which already exists?

File: web/modules/custom/hello_user/src/CurrentUser.php [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  }
+
+  /**
+   * Gets the cache tag for user.
+   * 
+   * @return string[]
+   *   The cache tag.
+   */
+  public function getUserCacheTags() {
+    $cacheTags = ['user:' . $this->currentUser->id()];
```
Comment (vivekpan27) [inline web/modules/custom/hello_user/src/CurrentUser.php:46]: Why create a new function for funtionality which already exists in core?

File: web/modules/custom/hello_user/src/Controller/HelloUser.php [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  }
+  
+  /**
+   * Renders a page to view user name.
+   * 
+   * @return array
+   *   An array suitable for showing content.
+   */
+  public function view() {
+    $this->content['name'] = $this->currentUser->getUserDisplayName();
```
Comment (vivekpan27) [inline web/modules/custom/hello_user/src/Controller/HelloUser.php:53]: 1/ Why assign this value to a variable?
2/ Why create a class property just for this?

File: web/modules/custom/custom_block_module/custom_block_module.module
Comment (vivekpan27) [inline web/modules/custom/custom_block_module/custom_block_module.module:1]: Why create this empty file?

**Review by vivekpan27 — COMMENTED** [review]
Feedback on custom code.
Rest looks ok.

File: web/modules/custom/custom_block_module/custom_block_module.module
Comment (Souvik-Inno) [reply web/modules/custom/custom_block_module/custom_block_module.module:1]: Removed.

File: web/modules/custom/hello_user/src/Controller/HelloUser.php (hunk 3 above)
Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/Controller/HelloUser.php]: 1/ Has assigned value to variable to test passing of more than one values to template. Removed now.
2/ Changed

File: web/modules/custom/hello_user/src/CurrentUser.php (hunk 1 above)
Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/CurrentUser.php:36]: Earlier has added code like:
`\Drupal::currentUser()->getAccountName();`
You added comment that it was not the proper way and check services and dependencies. Should it be done some other way?

Comment (vivekpan27) [reply web/modules/custom/hello_user/src/CurrentUser.php:36]: Check your first pull request.
There you have done it correctly :)

**Comment by vivekpan27** [comment]
@Souvik-Inno Please resolve the conflicts

File: web/modules/custom/hello_user/src/CurrentUser.php (hunk 1 above)
Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/CurrentUser.php]: Actually This file is same as the first pull request. I have not made any changes to this file ;)

Comment (vivekpan27) [reply web/modules/custom/hello_user/src/CurrentUser.php]: Yes, I see that now.
I think you are not using this service anymore.
So please remove the service and its definition.

Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/CurrentUser.php]: If I remove the service and its definition, then the module hello_user will not work as it depends on the service. It is actually for the task FT2023-339. Should I remove the entire module hello_user then?

Comment (vivekpan27) [reply web/modules/custom/hello_user/src/CurrentUser.php]: Please remove the service only not the module.
What you have done in this service is something which Drupal core service already does, so this is not adding any value to the existing code.
If you want to show that you have created a service, then please do something more meaningful in that.

Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/CurrentUser.php]: Okay removed the service.

File: web/modules/custom/hello_user/src/Controller/HelloUser.php [hunk 4]
```diff
@@ -2,62 +2,25 @@
...
    *   An array suitable for showing content.
    */
   public function view() {
-    $this->content['name'] = $this->currentUser->getUserDisplayName();
-    $var = $this->currentUser->getUserCacheTags();
-    
     return [
       '#theme' => 'hello_user',
-      '#content' => $this->content,
+      '#content' => \Drupal::currentUser()->getDisplayName(),
```
Comment (vivekpan27) [inline web/modules/custom/hello_user/src/Controller/HelloUser.php:21]: Not correct.
Use dependency injection.

File: web/modules/custom/hello_user/src/Controller/HelloUser.php [hunk 5]
```diff
@@ -2,62 +2,25 @@
...
       '#cache' => [
-        'tags' => $this->currentUser->getUserCacheTags(),
+        'tags' => ["user:" . \Drupal::currentUser()->id()],
```
Comment (vivekpan27) [inline web/modules/custom/hello_user/src/Controller/HelloUser.php:23]: Not correct. Use dependency injection.

**Review by vivekpan27 — CHANGES_REQUESTED** [review]

**Comment by vivekpan27** [comment]
@Souvik-Inno Did you run the code through PHP Codesniffer before creating the Pull Request?
Because if you would have, then these pending issues would have been detected.

File: web/modules/custom/hello_user/src/Controller/HelloUser.php (hunk 4 above)
Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/Controller/HelloUser.php]: Okay have fixed it now. Earlier was confused as I was unable to understand that you were telling to use core services in dependency injection. Now have understood and implemented it.

File: web/modules/custom/hello_user/src/Controller/HelloUser.php (hunk 5 above)
Comment (Souvik-Inno) [reply web/modules/custom/hello_user/src/Controller/HelloUser.php]: Fixed.

**Review by vivekpan27 — APPROVED** [review]

**Pull request merged**